prov/efa/src/rxr/rxr_ep.c

```diff
@@ -1840,14 +1841,12 @@ static inline fi_addr_t rdm_ep_determine_peer_address_from_efadv(struct rxr_ep *
 {
 	struct rxr_pkt_entry *pkt_entry;
 	struct efa_ep *efa_ep;
-	struct efa_cq *efa_cq;
 	struct efa_ep_addr efa_ep_addr = {0};
 	fi_addr_t addr;
 	union ibv_gid gid = {0};
 	uint32_t *connid = NULL;
 
-	efa_cq = container_of(ep->rdm_cq, struct efa_cq, util_cq.cq_fid);
-	if (!(efa_cq->flags & EFA_CQ_SUPPORT_EFADV_CQ)) {
+	if (ep->ibv_cq_ex_type != EFADV_CQ) {
```

Comment:
It looks the like the flags were set manually previously, so I don't know if there's a way to get the flags from the already created `ibv_cq_ex`

Wei suggested the enum approach and I followed it

Reply:
@wenduwan  I wonder if the flag EFA_CQ_SUPPORT_EFADV_CQ is really necessary.

The only purpose of it is to prevent later code to call `efadv_wc_read_sgid` on a `ibv_cq_ex` created by using `ibv_create_cq`.

I wonder why do we need to prevent this from happening? 

I understand such a call would fail because `ibv_cq_ex` created by `ibv_create_cq` does not support `efadv_wc_read_sgid`, but we should be able to a failure.

It would be nice to be able to get rid of the flag.

Reply:
> The only purpose of it is to prevent later code to call efadv_wc_read_sgid on a ibv_cq_ex created by using ibv_create_cq.
I wonder why do we need to prevent this from happening?

`EFA_CQ_SUPPORT_EFADV_CQ` is a runtime flag to indicate kernel module support for wide completion APIs. We introduced it because the installer kernel module could lag behind rdma-core APIs, such that we need to make sure both support wide completion:
1. For rdma-core APIs we just a macro, i.e. `HAVE_EFADV_CQ_EX`
1. For kernel model we cannot use a macro, but instead, we try to create edadv cq and if that fails we fallback to the normal `ibv_cq_ex`, i.e. a runtime flag `EFA_CQ_SUPPORT_EFADV_CQ`

Reply:
Tests showed that calling `efadv_wc_read_sgid` on a `ibv_cq_ex` created by using `ibv_create_cq` leads to a segfault, so we need to keep the enum.